third_party/amd/lib/TritonAMDGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp

```diff
@@ -381,26 +404,29 @@ fastPathComputeOffsets(ConversionPatternRewriter &rewriter, Location loc,
 }
 
 bool isColMajor(::llvm::ArrayRef<unsigned> order) {
-  assert(order.size() == 2 && (order[0] & ~1ul) == 0 &&
-         order[0] + order[1] == 1);
-  return order[0] == 0;
+  auto rank = order.size();
+  return order[0] == (rank - 2);
 }
 
 bool isKMajor(::llvm::ArrayRef<unsigned> order, int opIdx) {
-  if (order[0] + opIdx == 1)
+  auto rank = order.size();
+  if ((rank == 2) && (order[0] + opIdx == 1))
     return true;
-  else
-    return false;
+  if ((rank == 3) && (order[0] + opIdx == 2))
```

Comment:
The way I got over special-casing for rank==2 and rank==3 in SharedToDotOperandMMAv2 is to create a dummy 3D tensor out of the input tensor, do the codegen, an then throw away that 3D tensor, so we only have to think about 3D codegen rather than having if/else everywhere. Can we do the same here ?

Reply:
When I checked your PR I did not pay attention what you did in this file since I'm not familiar with mma layout. I think this is indeed a good way to get rid of if(rank == ) stuff as much as possible. However, given the size of this PR, can we do it in a future PR as a refactor?

Reply:
I think we should just do it now. We may forget about it if we delay refactoring to a future PR :)

Reply:
Fair enough. I'll find some time next week to refactor this part.